async-query-core/src/main/java/org/opensearch/sql/spark/scheduler/AsyncQueryScheduler.java

```diff
@@ -40,9 +40,11 @@ public interface AsyncQueryScheduler {
    * Temporarily disabling a job during maintenance or high-load periods - Allowing for easy
    * re-enabling of the job in the future
    *
-   * @param jobId The unique identifier of the job to unschedule
+   * @param asyncQuerySchedulerRequest The request containing updated job configuration
+   * @throws IllegalArgumentException if the job to be unscheduled doesn't exist
+   * @throws RuntimeException if there's an error during the unschedule process
    */
-  void unscheduleJob(String jobId);
+  void unscheduleJob(AsyncQuerySchedulerRequest asyncQuerySchedulerRequest);
```

Comment:
Is it right decision to pass AsyncQuerySchedulerRequest, which contains everything?
I think it would be confusing as very few attributes from this class is needed.

Reply:
as discussed offline, will keep signature as AsyncQuerySchedulerRequest for consistency of interface usage among multiple repos and updated the java doc here